### PR TITLE
Changes in the user guide layout

### PR DESCRIPTION
**Pull request by luisaFelixSalles**
This PR changes the user guide section 

- Add the tutorials subsection
- Adapt the index page of the main page and of the subsections  (all the cards design) 
- Change the sidebar: add captions for each subsection 

image
image
image
image

### CODE REVIEW

File: doc/source/user_guide/how-to/plotting.rst
Comment (PProfizi) [inline doc/source/user_guide/how-to/plotting.rst:1]: To move to tutorials

File: doc/source/user_guide/how-to/index.rst [hunk 1]
```diff
@@ -1,73 +1,47 @@
 .. _ref_how_to:
 
-How-tos
-~~~~~~~
+Accessing and enriching DPF capabilities
```
Comment (PProfizi) [inline doc/source/user_guide/how-to/index.rst:3]:
```suggestion
Enriching DPF capabilities
```

File: doc/source/user_guide/tutorials/index.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
+.. _ref_tutorials_index:
+
+Tutorials
+---------
+
+
+The following tutorials shows in details how to use basic features of PyDPF-Core
```
Comment (PProfizi) [inline doc/source/user_guide/tutorials/index.rst:7]:
```suggestion
The following tutorials show in details how to use basic features of PyDPF-Core
```

File: doc/source/user_guide/tutorials/index.rst [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+       :link: ref_tutorials_model
+       :link-type: ref
+       :text-align: center
+
+    .. grid-item-card:: DPF model
+       :link: ref_tutorials_model
+       :link-type: ref
+       :text-align: center
+
+**Store input data in DPF**
```
Comment (PProfizi) [inline doc/source/user_guide/tutorials/index.rst:27]:
```suggestion
**Load data in DPF**
```

File: doc/source/user_guide/tutorials/index.rst [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+       :link-type: ref
+       :text-align: center
+
+       LabelSpace
+
+       FieldsContainer
+
+       MeshContainer
+
+**Transform available data**
```
Comment (PProfizi) [inline doc/source/user_guide/tutorials/index.rst:65]:
```suggestion
**Transform data**
```

File: doc/source/user_guide/index.rst [hunk 5]
```diff
@@ -13,16 +13,51 @@ The goals of this section are to:
  - Describe the most-used DPF entities and how they can help you to access and modify solver data.
  - Provide simple how-tos for tackling the most common use cases.
```
Comment (PProfizi) [inline doc/source/user_guide/index.rst:14]: Mention the tutorials?

File: doc/source/user_guide/how-to/index.rst [hunk 6]
```diff
@@ -1,73 +1,47 @@
...
        :link: user_guide_server_context
        :link-type: ref
-       :width: 25%
        :text-align: center
 
        .. image:: ../images/drawings/entry_premium.png
 	   
 
-    .. card:: Use DPF XML Files
+    .. grid-item-card:: Use DPF XML Files
```
Comment (PProfizi) [inline doc/source/user_guide/how-to/index.rst:42]:
```suggestion
    .. grid-item-card:: Configure DPF plugins load
```

File: doc/source/user_guide/how-to/index.rst [hunk 7]
```diff
@@ -1,73 +1,47 @@
...
        :link-type: ref
-       :width: 25%
        :text-align: center
 
        .. image:: ../images/drawings/plugin-logo.png
           :width: 50%
 		  
 
-    .. card:: Use DPF Server package
+    .. grid-item-card:: Use DPF Server package
```
Comment (PProfizi) [inline doc/source/user_guide/how-to/index.rst:18]: It redirects to the Getting Started page for DPF Server, so remove it from here

File: doc/source/user_guide/how-to/index.rst [hunk 8]
```diff
@@ -1,73 +1,47 @@
...
-       :link-type: ref
-       :width: 25%
-       :text-align: center
-
-       .. image:: ../images/plotting/pontoon.png
-       .. image:: ../images/plotting/pontoon_strain.png
-
-
-    .. card:: Create custom operators
+    .. grid-item-card:: Create custom operators
```
Comment (PProfizi) [inline doc/source/user_guide/how-to/index.rst:9]:
```suggestion
    .. grid-item-card:: Create custom operators and plugins
```

File: doc/source/user_guide/tutorials/model.rst [hunk 9]
```diff
@@ -1,4 +1,4 @@
-.. _user_guide_model:
+.. _ref_tutorials_model:
```
[the remark this answers is not in the document]
Comment (luisaFelixSalles) [reply doc/source/user_guide/tutorials/model.rst:1]: yes

**Comment by PProfizi** [comment]
@luisaFelixSalles the doc step is [failing](https://github.com/ansys/pydpf-core/actions/runs/10963572060/job/30445332031?pr=1763#step:19:29) because the `create_sites_for_python_operators.*` scripts should not be moved.

**Comment by luisaFelixSalles** [comment]
> @luisaFelixSalles the doc step is [failing](https://github.com/ansys/pydpf-core/actions/runs/10963572060/job/30445332031?pr=1763#step:19:29) because the `create_sites_for_python_operators.*` scripts should not be moved.

You prefer if I change the reference in the other file or if I replace it ?

File: doc/source/user_guide/tutorials/index.rst [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+       Plotter
+
+       Animator
+
+**Export data**
+
+
+.. grid:: 3
+
+    .. grid-item-card:: Serialisation
```
Comment (PProfizi) [inline doc/source/user_guide/tutorials/index.rst:93]:
```suggestion
    .. grid-item-card:: Serialization
```
FYI: the English used is the american one, so with a "z".

File: doc/source/user_guide/tutorials/index.rst [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+    .. grid-item-card:: Process creation
+       :link: ref_tutorials_model
+       :link-type: ref
+       :text-align: center
+
+       Operators
+
+       Workflow
+
+    .. grid-item-card:: Results visualisation
```
Comment (PProfizi) [inline doc/source/user_guide/tutorials/index.rst:79]:
```suggestion
    .. grid-item-card:: Results visualization
```

File: doc/source/user_guide/index.rst [hunk 12]
```diff
@@ -11,18 +11,55 @@ computation, customization, and remote postprocessing accessible in Python.
 The goals of this section are to:
 
  - Describe the most-used DPF entities and how they can help you to access and modify solver data.
- - Provide simple how-tos for tackling the most common use cases.
+ - Provide simple tutorials for tackling the most common use cases.
+ - Explain how to customize DPF behavior
```
Comment (PProfizi) [inline doc/source/user_guide/index.rst:15]:
```suggestion
 - Explain how to customize DPF behavior.
```

File: doc/source/user_guide/index.rst [hunk 13]
```diff
@@ -11,18 +11,55 @@ computation, customization, and remote postprocessing accessible in Python.
...
+ - Present some systematic approaches to determine the cause of a malfunction or failure when using DPF
```
Comment (PProfizi) [inline doc/source/user_guide/index.rst:16]:
```suggestion
 - Present some systematic approaches to determine the cause of a malfunction or failure when using DPF.
```

**Review by PProfizi — APPROVED** [review]

**Pull request merged**